### PR TITLE
Add back BLS changes from orphaned blocks

### PR DESCRIPTION
**Pull request by potuz**
In case of a reorg, add back the BLS changes in the orphaned blocks.

### CODE REVIEW

File: beacon-chain/blockchain/head.go [hunk 1]
```diff
@@ -347,9 +348,9 @@ func (s *Service) notifyNewHeadEvent(
 	return nil
 }
 
-// This saves the attestations between `orphanedRoot` and the common ancestor root that is derived using `newHeadRoot`.
+// This saves the attestations and BLSToExecChanges between `orphanedRoot` and the common ancestor root that is derived using `newHeadRoot`.
```
Comment (terencechain) [inline beacon-chain/blockchain/head.go:351]: Looks nicer?

```suggestion
// This saves the Attestations and BLSToExecChanges between `orphanedRoot` and the common ancestor root that is derived using `newHeadRoot`.
```

File: beacon-chain/blockchain/head.go [hunk 2]
```diff
@@ -347,9 +348,9 @@ func (s *Service) notifyNewHeadEvent(
...
 // It also filters out the attestations that is one epoch older as a defense so invalid attestations don't flow into the attestation pool.
-func (s *Service) saveOrphanedAtts(ctx context.Context, orphanedRoot [32]byte, newHeadRoot [32]byte) error {
+func (s *Service) saveOrphanedOperations(ctx context.Context, orphanedRoot [32]byte, newHeadRoot [32]byte) error {
```
Comment (terencechain) [inline beacon-chain/blockchain/head.go:353]: Tbf this doesn't save all the operations, we should move slashing and exit to here in subsequent PR

**Review by terencechain — APPROVED** [review]

File: beacon-chain/blockchain/head.go (hunk 2 above)
Comment (potuz) [reply beacon-chain/blockchain/head.go:353]: Yep, was working on it when vacations started :)

File: beacon-chain/blockchain/head.go [hunk 3]
```diff
@@ -388,6 +389,15 @@ func (s *Service) saveOrphanedAtts(ctx context.Context, orphanedRoot [32]byte, n
 			}
 			saveOrphanedAttCount.Inc()
 		}
+		if orphanedBlk.Version() >= version.Capella {
+			changes, err := orphanedBlk.Block().Body().BLSToExecutionChanges()
+			if err != nil {
+				return errors.Wrap(err, "could not get BLSToExecutionChanges")
+			}
+			for _, c := range changes {
+				s.cfg.BLSToExecPool.InsertBLSToExecChange(c)
```
Comment (nisdas) [inline beacon-chain/blockchain/head.go:398]: What happens if the current branch already has these bls changes included ? Do we rely on our block packing to filter them out ?

Comment (potuz) [reply beacon-chain/blockchain/head.go:398]: Insertion is a noop in this case

**Pull request merged**